### PR TITLE
docs(hooks): summarise per-hook arguments on intro page (#4564)

### LINKED ISSUE #4564: Improving docs about available arguments in hooks.
## Description
I am currently teaching myself hooks (love them), but I find the docs not ideal in terms of variable access.

### Documentation page (if applicable)
https://docs.kedro.org/en/stable/hooks/introduction.html

On this page, I quickly understand that I can have access to those variables, although they could be described a bit better here, but I roughly get what they are.

I was then unable to find the accessible for all other hooks. In the end I went to the `specs.py` in the source code, and checked myself, but that was quite a bother.

```python
@hook_spec
def after_catalog_created(
    self,
    catalog: DataCatalog,
    conf_catalog: Dict[str, Any],
    conf_creds: Dict[str, Any],
    save_version: str,
    load_versions: Dict[str, str],
) -> None:
    pass
```

## Improvement
So my improvement suggestion would be to link the above described list of hooks to their actual specs, and maybe have a few more simple examples. If this is already described somewhere, I was unable to find it.

### PR DESCRIPTION
**Pull request by jbbqqf**
## Description

Closes #4564.

The Hooks [introduction page](https://docs.kedro.org/en/stable/extend/hooks/introduction/) currently lists hook names but not their arguments. Users have to either grep the source (`kedro/framework/hooks/specs.py`) or hop to the API reference page they didn't know existed. The intro page is also where new contributors land first, so it's the natural place to surface "which arguments can I receive in each hook?".

This PR addresses that:

- Adds a **per-hook argument summary table** under a new "Available arguments per hook" subsection.
- Calls out **pluggy's opt-in argument behaviour** so users know they don't have to declare every parameter — answering @datajoely's note in the issue thread that "the slightly unusual behaviour that not all arguments have to be provided should be covered too".
- Adds an explicit link to the `kedro.framework.hooks.specs` API reference from the intro paragraph, so the surfaceless API page is easier to find — addressing @maxschulz-COL's "at the very least I would link from the intro page to them".

The table is the load-bearing part. To stop it drifting silently when a spec adds or renames an argument, this PR also adds a regression test that parses the markdown table and compares it against `inspect.signature(...)` of every spec method.

## Development notes

**Files touched:**

- `docs/extend/hooks/introduction.md` — new "Available arguments per hook" subsection with the table; one-line tweak to the existing "full specifications" paragraph to link the API page.
- `tests/framework/hooks/test_manager.py` — adds `test_hook_args_doc_table_matches_specs` next to the existing parameterised hook-spec tests, so the canonical list of hook arguments has a single home.
- `RELEASE.md` — changelog bullet under `## Documentation changes`.

**The regression test fails on `origin/main`** (the section it parses doesn't exist there) and **passes on this branch**. It also fails if a future change adds/removes an argument on a spec without updating the docs table — that's the contract being preserved.

**Build verification:** `mkdocs build --strict` succeeds; the new internal anchor links resolve.

## Reproduce BEFORE/AFTER yourself (copy-paste)

A reviewer can verify in under a minute by pasting the block below.

```bash
# --- one-time setup ---
git clone https://github.com/kedro-org/kedro.git /tmp/repro && cd /tmp/repro
python -m venv .venv && source .venv/bin/activate
pip install -e ".[test]"

# --- BEFORE (origin/main) ---
git checkout origin/main
git fetch https://github.com/jbbqqf/kedro.git feat/4564-hooks-args-table
# Borrow only the new test from the PR branch so we can run it against main:
git checkout FETCH_HEAD -- tests/framework/hooks/test_manager.py
pytest tests/framework/hooks/test_manager.py::test_hook_args_doc_table_matches_specs -v --no-cov
# Expected: FAILED — "Section '### Available arguments per hook' not found"
git checkout origin/main -- tests/framework/hooks/test_manager.py

# --- AFTER (this PR) ---
git checkout FETCH_HEAD
pytest tests/framework/hooks/test_manager.py::test_hook_args_doc_table_matches_specs -v --no-cov
# Expected: PASSED — table is in sync with the actual hook spec signatures
```

Optional manual check of the rendered docs:

```bash
pip install -e ".[docs]"
mkdocs serve
# Open http://127.0.0.1:8000/extend/hooks/introduction/
# Expected: a new "Available arguments per hook" subsection with a table
# listing each hook and the arguments it receives, plus a link to the
# kedro.framework.hooks.specs API reference.
```

## What I ran locally

- `pytest tests/framework/hooks/ -v --no-cov` → 16/16 passed (the 12 existing parameterised cases + 1 new doc-sync test + 3 other manager tests).
- `pytest tests/framework/hooks/test_manager.py::test_hook_args_doc_table_matches_specs -v --no-cov` on `origin/main` → 1 failed (assertion on missing section) — confirming the regression test catches the gap.
- `ruff check tests/framework/hooks/test_manager.py docs/extend/hooks/introduction.md` → all checks passed.
- `ruff format --check tests/framework/hooks/test_manager.py` → already formatted.
- `mkdocs build --strict` → built in 42.69 s, no broken links.

## Edge cases tested

| # | Scenario | Input | Expected | Verified by |
|---|----------|-------|----------|-------------|
| 1 | Table in sync with specs | current PR state | test passes | `test_hook_args_doc_table_matches_specs` |
| 2 | Section deleted from intro page | section header removed | test fails with explicit "section not found" message | manual: deleted the heading, reran the test |
| 3 | Argument added to a spec without updating docs | added a fake arg to `before_node_run` locally | test fails with `documented={...} actual={...}` diff | manual: patched specs.py, reran the test, then reverted |
| 4 | Argument renamed on a spec | renamed `dataset_name` to `name` locally on `DatasetSpecs` | test fails | manual: patched specs.py, reran, reverted |

## Risk / blast radius

- Documentation-only change; no runtime behaviour modified.
- The regression test only reads two files (`introduction.md` and `kedro.framework.hooks.specs`) and uses standard-library `inspect` and `re` — no new third-party dependencies.
- The test's regex parser is intentionally narrow (anchored on the exact section heading and the backticked-hook-name pattern) so prose changes around the table won't break it.
- Anchor links point at the existing `kedro.framework.hooks.specs` autodoc page already shipped in `docs/api/framework/kedro.framework.hooks.md`.

## Release note

```release-note
docs(hooks): summarise per-hook arguments on the introduction page and link the API reference
```

## Developer Certificate of Origin

All commits are signed off (`Signed-off-by: J-Baptiste Braun <jbaptiste.braun@gmail.com>`).

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes (this PR *is* the documentation change)
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes (`test_hook_args_doc_table_matches_specs`)
- [x] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team — N/A, docs-only and only touches `kedro.framework.hooks.specs`-derived content

---

*PR drafted with assistance from Claude Code. The change was reviewed manually against `kedro/framework/hooks/specs.py` and the upstream pluggy docs cited above. The reproducer block above was used during development; it is the same one a reviewer can paste verbatim.*

### CODE REVIEW

File: tests/framework/hooks/test_manager.py [hunk 1]
```diff
@@ -77,6 +81,76 @@ def test_hook_manager_can_call_hooks_defined_in_specs(
...
+        / "introduction.md"
+    )
+    text = docs_path.read_text(encoding="utf-8")
+
+    # Locate the "Available arguments per hook" section and parse the markdown
+    # table that follows it. The parser is intentionally narrow: it only matches
+    # the rows whose first column is a backticked hook name, so prose around
+    # the table is ignored.
+    section = re.search(
+        r"###\s+Available arguments per hook\s*\n(.*?)(?:\n##\s|\Z)",
```
Comment (rashidakanchwala) [inline tests/framework/hooks/test_manager.py:107]: **Suggestion:** The regex stops at `\n##\s` (two hashes) but not `\n###\s` (three hashes), so the capture includes `### CLI Hooks` and everything beyond it until `## Hook implementation`. In practice this is harmless since that section has no matching table rows, but the intent is clearly to scope to just `### Available arguments per hook`. A tighter stop condition:

```python
r"###\s+Available arguments per hook\s*\n(.*?)(?:\n###\s|\n##\s|\Z)"
```

**Review by rashidakanchwala — COMMENTED** [review]
Thank you for this contribution, @jbbqqf! The table is exactly what the intro page was missing, the regression test is a great idea, and the PR description + reproduce steps are exceptionally thorough. One small suggestion below — nothing blocking.

## Kedro PR Review
> Generated with `review-kedro-pr` skill.

### Overview
- **Critical:** 0 | **Suggestions:** 1

### Critical (must fix before merge)
No issues found.

### Suggestions (consider improving)
- [ ] `tests/framework/hooks/test_manager.py:107` — regex section boundary bleeds into `### CLI Hooks`; add `\n###\s` as a stop condition

### Notes
- **General review:** Docs content is relevant, well-placed, and not repeated elsewhere in the hooks docs.
- **PR checklist:** All boxes ticked. RELEASE.md entry present under `## Documentation changes`. ✓
- **API usage:** Docs-only change. No backward compatibility or Kedro-Viz concerns.

**Review by rashidakanchwala — APPROVED** [review]

**Comment by lrcouto** [comment]
Hello @jbbqqf ! The text itself looks nice, I'd just apply the suggestions from the language linter. There's a run-on sentence and some weasel words. Other than that, I'd say it's good to go.

**Comment by merelcht** [comment]
Hi @jbbqqf, are you still interested in getting this PR merged? Let the team know if you need any assistance.

**Review by lrcouto — APPROVED** [review]
Fixed the Vale issues, text is good otherwise. Thank you!
